### PR TITLE
Release preparation for v0.17.0

### PR DESCRIPTION
**Pull request by lexnv**
Release preparation for v0.17.0

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,20 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.17.0] - 2022-02-04
+
+### Added
```
Comment (jsdw) [inline CHANGELOG.md:11]: Nit: there's some inconsistency over "do we have a newline between the title and list" in this doc; let's add a newline here

Comment (lexnv) [reply CHANGELOG.md:11]: Good catch, fixed in the latest revision

File: test-runtime/Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ sp-runtime = "5.0.0"
 codec = { package = "parity-scale-codec", version = "2", default-features = false, features = ["derive", "full", "bit-vec"] }
 
 [build-dependencies]
-subxt = { path = "../subxt", version = "0.16.0" }
```
Comment (jsdw) [inline test-runtime/Cargo.toml:12]: let's remove the `version` entirely from this to save us a little future trouble.

File: macro/Cargo.toml [hunk 3]
```diff
@@ -27,10 +27,10 @@ quote = "1.0.8"
 syn = "1.0.58"
 scale-info = "1.0.0"
 
-subxt-codegen = { path = "../codegen", version = "0.16.0" }
+subxt-codegen = { path = "../codegen", version = "0.17.0" }
 
 [dev-dependencies]
 pretty_assertions = "1.0.0"
-subxt = { path = "../subxt", version = "0.16.0" }
```
Comment (jsdw) [inline macro/Cargo.toml:34]: Let's remove `version` from this; if I read it right, doing so may mean we no logner need to use `cargo hack` to publish this crate as `cargo publish` will strip the dependency automatically!

**Review by jsdw — APPROVED** [review]
LGTM :)

File: CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+
+- Get event context on EventSubscription ([#423](https://github.com/paritytech/subxt/pull/423))
+
+### Changed
```
Comment (niklasad1) [inline CHANGELOG.md:15]: isn't the jsonrpsee change missing from here or excluded from the release?

**Review by niklasad1 — APPROVED** [review]
LGTM, one question about the CHANGELOG

File: CHANGELOG.md (hunk 4 above)
Comment (jsdw) [reply CHANGELOG.md:15]: Ah good point; the default sort order of PRs is annoying; perhaps it's worth noting in the release doc to filter by `is:pr sort:updated-desc is:closed` to see PRs in order of when they were updated.

Everything else looks good I think; just that one that's missing

**Review by jsdw — APPROVED** [review]

File: CHANGELOG.md (hunk 4 above)
Comment (niklasad1) [reply CHANGELOG.md:16]: yupp, makes sense

**Pull request merged**